### ISSUE TITLE
[MU4 Issue] Accessibility: toggle note input / normal modes does not read the change

### ISSUE THREAD

**Issue by MarcSabatella** [musescore/MuseScore#12966]
**Describe the bug**
When pressing "N" to toggle note input mode on or off, the screen reader should read the name of the mode.

**To Reproduce**
1. Start screen reader
2. Press "N" to toggle note input mode
3. See error: the screen reader does not tell you which mode you are in

**Expected behavior**
Screen reader should read the name of the mode as you enter it

**Platform information**
 - OS: Windows, Linux

**Additional context**
Originally it did not for MU3 either, but this feature was added around 3.5 in https://github.com/musescore/MuseScore/pull/6173

**Comment by shoogle** [comment]
The specific commit was https://github.com/musescore/MuseScore/pull/6173/commits/3960396bf53c41637fed346b3d9b96d89f1cc339. The implementation was a hack made necessary by MU3's accessibility system.

We're trying to do things properly in MU4, but I'm not sure if there is a proper way to do this. We want to send generic status messages like:

- _"Note input mode"_ (when you enter that mode)
- _"Normal mode"_ (when you leave note input mode)
- _"Download complete"_

Unlike most screen reader messages, these aren't related to navigation (i.e. they are not caused by a change in focus, value, or selection). Unfortunately, **I don't know of a way to send messages that are not related to navigation**. If it's not possible then we'll have to resort to a similar hack where we _pretend_ that there has been a navigational change just to get the screen reader to say something.

**Comment by shoogle** [comment]
A suitable hack might be:

- Change the name or value of the focus object to _"Note input mode"_ (or whatever).
- Select a dummy object that has the name or value _"Note input mode"_ (or whatever).

The second option is probably more reliable (screen readers sometimes ignore a change in name or value but they rarely ignore a change in focus). However, the dummy object needs to have the same parent in the accessibility tree as the real object or there is a risk that the screen reader will say the parent's name as well as the dummy object's name.

Of course we wouldn't really change the name or value of anything in the score/UI. We would simply send an accessible event to report that such a change has happened, even though it hasn't really.

The same hack might help with #10228, although in that case it might be preferable to really change the name of the selected object to _"Range selection..."_ instead of just pretending to do this.

**Comment by MarcSabatella** [comment]
If it turns out to be simpler to fix the "get location" command Shift+L, which currently does not work (see https://github.com/musescore/MuseScore/issues/12100, then perhaps *it* could be made to also read the current mode somehow, and it might feel less hacky there.

**Comment by Eism** [comment]
Yeah, I also didn't find a way to send messages that are not related to navigation. Implemented adding a command to the beginning of the name of the current focused element and re-voicing its name.

Maybe in the future we will find a better solution.
